src/System.Device.Gpio/System/Device/Gpio/Drivers/SysFsDriver.Linux.cs

```diff
@@ -64,14 +85,15 @@ protected internal override void OpenPin(int pinNumber)
         /// <param name="pinNumber">The pin number in the driver's logical numbering scheme.</param>
         protected internal override void ClosePin(int pinNumber)
         {
-            string pinPath = $"{GpioBasePath}/gpio{pinNumber}";
+            int pinOffset = pinNumber + _pinOffset;
+            string pinPath = $"{GpioBasePath}/gpio{pinOffset}";
             // If the directory doesn't exist, this becomes a no-op since the pin was closed already.
             if (Directory.Exists(pinPath))
             {
                 try
                 {
                     SetPinEventsToDetect(pinNumber, PinEventTypes.None);
-                    File.WriteAllText(Path.Combine(GpioBasePath, "unexport"), pinNumber.ToString());
+                    File.WriteAllText(Path.Combine(GpioBasePath, "unexport"), pinOffset.ToString());
```

Comment:
>ToString() [](start = 88, length = 10)

nit: maybe better to pass CultureInfo.InvariantCulture to ToString?

Reply:
pin numbers are integer and would not be scaled so high, do you think its better to have CultureInfo.InvariantCulture or it parses faster with that culture?

Reply:
Please use InvariantCulture. it is faster.

Reply:
sure, thank you